lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -188,8 +195,46 @@ public DataLoadingRequestBuilder setInputFormatClass(
     return this;
   }
 
+  /**
+   * Sets the path of the folder where the data is.
+   * Set to ANY the evaluator that will be able to load this data.
+   *
+   * @deprecated since 0.12. Should use instead
+   *             {@link DataLoadingRequestBuilder#addDataPartition(DataPartition)}
+   *             or {@link DataLoadingRequestBuilder#addDataPartitions(List)}
+   * @param inputPath
+   *          the input path
+   * @return this
+   */
+  @Deprecated
   public DataLoadingRequestBuilder setInputPath(final String inputPath) {
-    this.inputPath = inputPath;
+    this.partitions = new ArrayList<>(Arrays.asList(new DataPartition(inputPath, DataPartition.ANY)));
```

Comment:
setInputPath is the old method, which we are deprecating now. In order to maintain the previous functionality in old code, setting the input path should just set one partition. That is why I'm creating a new list. setXXX should have a different semantic than addXXX. I believe this is the correct way to do it.
Old code is not using addDataPartition neither addDataPartitions as they are both new methods. 
This method should be removed in the future, and only the add ones should be used. I believe that adding a comment in this one could clarify that we are overriding everything, not adding. Do you agree?